### ISSUE TITLE
Interactive path

### ISSUE THREAD

**Issue by claucece** [otrv4/pidgin-otrng#88]
Ensure that the test case in https://github.com/otrv4/pidgin-otrng/wiki/Test-Case---Interactive-path works correctly. In the process of doing that, fill in the documentation in the wiki with more details where necessary.

* I check that I have the correct persisted values: client profile, prekey profile, private key, forging key, prekey messages, fingerprints, and shared prekey file.
* I open pidgin with the otr-ng plugin enabled.
* I activate my account and enter my password (if needed)
* I successfully log in
* When I log in, either prekey messages get published or a message says that we have enough prekey messages
* I click over a peer to whom I want to have a conversation with: a conversation window pops up
* When I click on the OTR button and the other participant is online, an OTR conversation is started by saying: "Attempting to start a private conversation" and "Private conversation started."
* I am able to send encrypted messages
* The other participant is able to receive my message and send messages as well

**Comment by giovaneliberato** [comment]
Hey, I found a inconsistency which is worth looking since it's related to user feedback. The change seems to be introduced by c276bfa7 

There are two conflicting messages showing up in the conversation window displaying OTR conversation status updates. In the first case, one saying "Not Private" and the other says "Private". In the second case, it shows "Not Private" and the usual "Private conversation started".

This seems to be a intermediate state that never shows up independently as far as I tested. Also there's the "Finished" state that I never managed to display in a conversation.

I'm not sure if we should fix this logic or remove this at all, since it seems old and buggy and we already have a label displaying the OTR conversation status. Thoughts @claucece @DrWhax @olabini ? 

Here's the code https://github.com/otrv4/pidgin-otrng/blob/master/gtk-dialog.c#L2765

First case - steps to reproduce:
- Start Pidgin
- Double click on "bob@localhost" to open a conversation window with Bob
- Click the "OTR" menu and then Click on "Start private conversation"
- Switch to bob's conversation window with alice@

![bug1](https://user-images.githubusercontent.com/2964631/46628927-8de2c180-cb15-11e8-875d-b1860541c298.png)

<br>

Second case - steps to reproduce:
- Start Pidgin
- Double click on "bob@localhost" to open a conversation window with Bob
- Click the "OTR" menu and then Click on "Start private conversation"
- Switch to bob's conversation window with alice@
- Click the "OTR" menu and then Click on "End private conversation"
- Switch to alices's conversation window with bob@
- Click the "OTR" menu and then Click on "Start private conversation"

![captura de tela de 2018-10-08 16-19-06](https://user-images.githubusercontent.com/2964631/46629046-e914b400-cb15-11e8-8424-cf52ba2c0dc5.png)

**Comment by claucece** [comment]
Hey!

Thanks for looking at this. 

I have never seen this message: 'The privacy status of the current conversation is now: Private' nor 'The privacy status of the current conversation is now: Not Private' in any case.  The status is only show in the bar for me. 

> The change seems to be introduced by c276bfa

I don't see where this was introduced.. can you point the line and file? This is a huge commit that changes a lot of things..

If you are referring to this:

  switch (current_level) {
  case TRUST_NOT_PRIVATE:
    status = _("Not Private");
    break;
  case TRUST_UNVERIFIED:
    status = _("Unverified");
    break;
  case TRUST_PRIVATE:
    status = _("Private");
    break;
  case TRUST_FINISHED:
    status = _("Finished");
    break;
  }

from what I remember that is used to changed the appearance of buttons and other stuff, so I'll say don't remove that. 

> This seems to be a intermediate state that never shows up independently as far as I tested. Also there's the "Finished" state that I never managed to display in a conversation.

The "Finished" state is reached on the menu bar when the other party ended the conversation: 

<img width="433" alt="screen shot 2018-10-09 at 2 55 02 am" src="https://user-images.githubusercontent.com/8270508/46649242-bba82480-cb6e-11e8-89c7-1aaa68e850e0.png">

**Comment by claucece** [comment]
I pushed something @giovaneliberato .. Now, at least I see the status.. check if it solves for you :).

**Comment by giovaneliberato** [comment]
> I don't see where this was introduced.. can you point the line and file? This is a huge commit that changes a lot of things..

I found it hard to track too, but the point I want to make is that this is a very old change that we inherit.
<br>

> I have never seen this message: 'The privacy status of the current conversation is now: Private' nor 'The privacy status of the current conversation is now: Not Private' in any case. The status is only show in the bar for me.

Hmm, that's weird. 
For me now it's being displayed three times for the use case 1 I described above

![weird](https://user-images.githubusercontent.com/2964631/46679844-2df42580-cbbe-11e8-8926-e36b31fda7d4.png)
<br>

> from what I remember that is used to changed the appearance of buttons and other stuff, so I'll say don't remove that.

Actually the logic for the OTR button is implemented here in [gtk-dialog.c:1232](https://github.com/otrv4/pidgin-otrng/blob/master/gtk-dialog.c#L1232). 

This switch case located at [gtk-dialog.c#L2743](https://github.com/otrv4/pidgin-otrng/blob/master/gtk-dialog.c#L2743) is used only for this message that is shown in the message box. Line `2766` contains the template being used in these messages. 

So my suggestion to remove this is because:
1) We already have two places informing the current status: 
- The OTR button
- The messages `Unverified/Private conversation started. Your client[..]` and `Private conversation lost.` that shows up in the conversation window.
2) It seems like it working inconsistently between us two, maybe someone else can try it.
3) It contains links that #95 aims to remove

What do you think @claucece?

**Comment by claucece** [comment]
Hey!

> Actually the logic for the OTR button is implemented here in gtk-dialog.c:1232.
> This switch case located at gtk-dialog.c#L2743 is used only for this message that is shown in the message box. Line 2766 contains the template being used in th

I'm not referring to the OTR button, but to the buttons of the OTR menu. This kind of trust level is used in several places to define which button gets active, etc. But I see for this case is only for the information. 

> The messages Unverified/Private conversation started. Your client[..] and Private conversation lost. that shows up in the conversation window.

I actually think that in this case it seems to be ok. In the course of a long conversation, I don't think an user is going to only look at the bottom toolbar to check the status. Maybe we should unified them into a more explicit 'Unverified/Private conversation started.'.. I never liked those texts anyhow.

> It seems like it working inconsistently between us two, maybe someone else can try it.

Is it still working inconsistency after commit d452471d889652a1dc66846bf2bf835106dcb64b ? It is showing and working ok now for me.. not for you?

> It contains links that #95 aims to remove

Those can be easily removed. 

Thanks!

**Comment by claucece** [comment]
So, after thinking for a while.. I think it makes sense to remove them. But I think we should create better texts that unify the ideas of 'Private conversation started.  Your client is not logging this conversation and The privacy status of the current conversation is now: Private'. Right now, both texts are not clear on what they mean. I'll push a commit removing the hyperlink and then, if you can, @giovaneliberato you can remove them. Thanks!

**Comment by giovaneliberato** [comment]
> So, after thinking for a while.. I think it makes sense to remove them. But I think we should create better texts that unify the ideas of 'Private conversation started. Your client is not logging this conversation and The privacy status of the current conversation is now: Private'. Right now, both texts are not clear on what they mean. 

Agree! I'll remove these messages and we work on improving the texts.

> Is it still working inconsistency after commit d452471 ? It is showing and working ok now for me.. not for you?

No :( 
For me it is showing multiple times as in the screen shot. I'm wondering why we'd see such differences.

**Comment by MauroVelasco** [comment]
Hey, found some inconsistencies  with the flow of this test case, the suggested flow was updated in the Test Case Wiki, the modifications are related with:
- Add assumptions
- Specify test case
- Specify goal and conclusion

**Comment by claucece** [comment]
Awesome @MauroVelasco ! Thanks so much!

**Comment by MauroVelasco** [comment]
After complete the flow the following files are not created:
- otr4.client_profile
- otr4.forging_key
- otr4.prekey_messages
- otr4.prekey_profile
- otr4.private_key

**Comment by MauroVelasco** [comment]
Pidgin is closed unexpectedly in this step:
`Click on "Start private conversation"`
